vscode-extension/src/extension.ts

```diff
@@ -233,22 +223,64 @@ async function createNewAIConfig(
   const fileContentBuffer = await vscode.workspace.fs.readFile(fileContentPath);
   const initialContent = fileContentBuffer.toString();
 
-  const doc = await vscode.workspace.openTextDocument({
-    content: initialContent,
-    language: mode,
+  const workspaceUri = vscode.workspace.workspaceFolders
+    ? vscode.workspace.workspaceFolders[0].uri
+    : null;
+
+  // Find the first available untitled file name to suggest as default
+  let firstAvailableUntitledName: string | null = null;
+  let i = 0;
+  while (firstAvailableUntitledName === null) {
+    const fileName = `untitled${i === 0 ? "" : "-" + i}.aiconfig.${mode}`;
+    const filePath = workspaceUri
+      ? path.join(workspaceUri.path, fileName)
+      : fileName;
+
+    if (fs.existsSync(filePath)) {
+      i++;
+    } else {
+      firstAvailableUntitledName = fileName;
+    }
+  }
+
+  const newConfigName = await vscode.window.showInputBox({
+    prompt: "Enter a name for the new AIConfig file",
+    value: firstAvailableUntitledName,
+    validateInput: (input) => {
+      const nameValidationError = validateConfigName(input, mode);
+      if (nameValidationError) {
+        return nameValidationError;
+      }
+
+      return fs.existsSync(input) ? "File already exists" : null;
```

Comment:
Can we just put this inside of the `validateConfigName()` method to keep it cleaner?